src/main/java/com/duckblade/osrs/dpscalc/calc/gearbonus/ObsidianGearBonus.java

```diff
@@ -0,0 +1,86 @@
+package com.duckblade.osrs.dpscalc.calc.gearbonus;
+
+import com.duckblade.osrs.dpscalc.calc.EquipmentItemIdsComputable;
+import com.duckblade.osrs.dpscalc.calc.compute.ComputeContext;
+import com.duckblade.osrs.dpscalc.calc.compute.ComputeInputs;
+import com.duckblade.osrs.dpscalc.calc.exceptions.DpsComputeException;
+import com.duckblade.osrs.dpscalc.calc.model.GearBonuses;
+import com.google.common.collect.ImmutableSet;
+import lombok.RequiredArgsConstructor;
+import net.runelite.api.EquipmentInventorySlot;
+import net.runelite.api.ItemID;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Map;
+import java.util.Set;
+
+@Singleton
+@RequiredArgsConstructor(onConstructor = @__(@Inject))
+public class ObsidianGearBonus implements GearBonusComputable
+{
+	private static final Set<Integer> OBSIDIAN_WEAPONS = ImmutableSet.of(
+		// Is the staff affected? Wiki is unclear. Will anybody ever use it? We'll find out.
+		// ItemID.TOKTZMEJTAL,
+	ItemID.TOKTZXILAK,
+	ItemID.TOKTZXILEK,
+	ItemID.TZHAARKETEM,
+	ItemID.TZHAARKETOM,
+	ItemID.TZHAARKETOM_T
+	);
+
+	private static final Set<Integer> BERSERKER_NECKLACES = ImmutableSet.of(
+		ItemID.BERSERKER_NECKLACE,
+		ItemID.BERSERKER_NECKLACE_OR
+	);
+
+	private static final Set<Integer> OBSIDIAN_HELMETS = ImmutableSet.of(
+		ItemID.OBSIDIAN_HELMET
+	);
+
+	private static final Set<Integer> OBSIDIAN_PLATEBODIES = ImmutableSet.of(
+		ItemID.OBSIDIAN_PLATEBODY
+	);
+
+	private static final Set<Integer> OBSIDIAN_PLATELEGS = ImmutableSet.of(
+		ItemID.OBSIDIAN_PLATELEGS
+	);
+
+	private final EquipmentItemIdsComputable equipmentItemIdsComputable;
+
+	@Override
+	public boolean isApplicable(ComputeContext context)
+	{
+		try
+		{
+			boolean castingSpell = context.get(ComputeInputs.ATTACK_STYLE).isManualCast();
+			return OBSIDIAN_WEAPONS.contains(context.get(equipmentItemIdsComputable).get(EquipmentInventorySlot.WEAPON)) &&
+					!castingSpell;
+		}
+		catch (DpsComputeException e)
```

Comment:
Don't catch DpsComputeExceptions during the compute flow.

Reply:
Fair enough, I added this in anticipation of null spells as in the other PR so it can be removed without complaint.